libPhoneNumber/Internal/NBRegExMatcher.h

```diff
@@ -0,0 +1,28 @@
+//
+//  NBRegExMatcher.h
+//  libPhoneNumber
+//
+//  Created by Paween Itthipalkul on 11/29/17.
+//  Copyright © 2017 Google LLC. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class NBPhoneNumberDesc;
+
```

Comment:
Why are we pulling out what is essentially a single method into its own class?

Reply:
This is to match the canonical libPhoneNumber (Java) implementation (though the Java version is an interface).

https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/internal/MatcherApi.java

When we talked to the libPhoneNumber maintainer at Google about porting (our team is doing the same porting to JavaScript libPhoneNumber), they prefer a direct port. Even though the iOS fork is not maintained by the libPhoneNumber, I feel like we should try to be as close to the canonical code as possible with Objective-C coding code idiom in mind.

In this case, it looks like NBPhoneNumberUtil is doing a bunch of regex stuff that shouldn't be in there, so I can pull this method into NBPhoneNumberUtil if you feel strongly about it.

If we were to pull this into NBPhoneNumberUtil, I'll remove the NBRegularExpressionCache as well.

Reply:
hmm.. I'll look at trying to clean this up later.